### PR TITLE
Cleanup: Silence Visual Studio 2017 warnings.

### PR DESCRIPTION
**Pull request by KevinHopps**

### CODE REVIEW

**Comment by MarcelRaad** [comment]
I've also been bugged by those warnings. There's already #44 from April.

**Comment by Lastique** [comment]
Looks like you introduced tabs with your edits. Please, convert to spaces.

**Comment by jeking3** [comment]
@KevinHopps would you be able to convert the tabs to spaces in your pull request to align with the current code style?  Thanks.

File: src/unique_path.cpp [hunk 1]
```diff
@@ -111,7 +111,9 @@ void system_crypt_random(void* buf, std::size_t len, boost::system::error_code*
 
   if (!errval)
   {
-    BOOL gen_ok = ::CryptGenRandom(handle, len, static_cast<unsigned char*>(buf));
+    // Use static cast to silence Visual Studio 2017
+    // warning C4267: 'argument': conversion from 'size_t' to 'DWORD', possible loss of data
+    BOOL gen_ok = ::CryptGenRandom(handle, static_cast<DWORD>(len), static_cast<unsigned char*>(buf));
```
Comment (jeking3) [inline src/unique_path.cpp]: @Lastique here's another random implementation that is not ready for Windows UWP.

Comment (jeking3) [reply src/unique_path.cpp]: Should we be using `static_cast` or should we be using `boost::numeric_cast` to ensure proper handling of math here (and the other instances I see in the other file).

Comment (KevinHopps) [reply src/unique_path.cpp]: numeric_cast seems appropriate. I will make that change.

**Comment by alexeikh** [comment]
Now in 2 of 3 places the comment says "Use static cast", but the code uses `numeric_cast`.

**Comment by jeking3** [comment]
In general, the comments in the code here are unnecessary.

File: src/windows_file_codecvt.cpp [hunk 2]
```diff
@@ -59,8 +60,10 @@
     UINT codepage = AreFileApisANSI() ? CP_ACP : CP_OEMCP;
 
     int count;
+    // Use static cast to silence Visual Studio 2017:
+    // warning C4244: 'argument': conversion from '__int64' to 'int', possible loss of data
     if ((count = ::WideCharToMultiByte(codepage, WC_NO_BEST_FIT_CHARS, from,
-      from_end - from, to, to_end - to, 0, 0)) == 0)
+       boost::numeric_cast<int>(from_end - from), to, boost::numeric_cast<int>(to_end - to), 0, 0)) == 0)
```
Comment (alexeikh) [inline src/windows_file_codecvt.cpp]: Sorry for nitpicking again. This line has 1 extra indentation space, compared to the original line.

Also, this line has become very long. I'd rewrite the whole fragment, for example, as:

```c++
int count = ::WideCharToMultiByte(
  codepage, WC_NO_BEST_FIT_CHARS,
  from, boost::numeric_cast<int>(from_end - from),
  to, boost::numeric_cast<int>(to_end - to),
  0, 0);
if (count == 0)
  return error;  // conversion failed
```

File: src/windows_file_codecvt.cpp [hunk 3]
```diff
@@ -40,7 +41,7 @@
 
     int count;
     if ((count = ::MultiByteToWideChar(codepage, MB_PRECOMPOSED, from,
-      from_end - from, to, to_end - to)) == 0) 
+      boost::numeric_cast<int>(from_end - from), to, boost::numeric_cast<int>(to_end - to))) == 0)
     {
       return error;  // conversion failed
     }
```
Comment (alexeikh) [inline src/windows_file_codecvt.cpp]: The expression in `if` is now very long. Suggesting to rewrite as:

```c++
int count = ::MultiByteToWideChar(
  codepage, MB_PRECOMPOSED,
  from, boost::numeric_cast<int>(from_end - from),
  to, boost::numeric_cast<int>(to_end - to));
if (count == 0)
  return error;  // conversion failed
```

**Comment by Lastique** [comment]
Seems to be already fixed by d88945564a4ed037474f9a5952b87f4f9244d4ba.

**Pull request closed without merging**